### ISSUE TITLE
Support custom scrobblers

### ISSUE THREAD

**Issue by ijanos** [newgiin/cloudplayer-scrobbler#32]
Please provide a way in the options page to set a custom scrobbler api endpoint.

**Comment by newgiin** [comment]
You mean like scrobble to something other than last.fm? Does it have an identical API? I'm not likely to implement it myself, but if you're up to it, you can try changing this file to point to something else: https://github.com/newgiin/cloudplayer-scrobbler/blob/0e000304f07d123fb533db67247e43d561d00950/js/lastfm.js

**Comment by ijanos** [comment]
Yeah there are API identical alternative implementations, most prominently Libre.FM 

What I am asking for is basically a text input box on the options page where I can change the value the API_ROOT variable.

**Comment by burkemw3** [comment]
A challenge with custom API endpoints is the chrome extension permissions ([ref][https://developer.chrome.com/extensions/xhr#requesting-permission]). I wouldn't be excited about this extension requesting permissions to all domains.

Supporting the default libre.fm end point is probably feasible (I think `https://turtle.libre.fm` instead of `ws.audioscrobbler.com`).

Libre.fm [docs do have an interesting workaround](https://git.gnu.io/foocorp/librefm/wikis/Using_turtle#step-two): use your hosts file to point `*.audioscrobbler.com` to `turtle.libre.fm`.

**Comment by burkemw3** [comment]
Looks like chrome supports requesting permissions during run-time [docs](https://developer.chrome.com/extensions/permissions). It looks possible to make `http://*/` an optional permissions and request access to specific end points as needed. So, my last message probably isn't much of a challenge.
